src/comm.c

```diff
@@ -345,23 +360,32 @@ void game_loop_unix(int control) {
         if (doormanDesc && FD_ISSET(doormanDesc, &in_set)) {
             do {
                 Packet p;
-                char buffer[4096 * 4];
+                char buffer[UMAX(PACKET_MAX_PAYLOAD_SIZE, 4096 * 4)];
```

Comment:
Bit concerned about the size of this. It's always going to be 16384. But it's then writing into d->inbuf in read_from_descriptor(), which is 4*MAX_INPUT_LENGTH.    Plus, doorman itself in Xania.send_to_mud won't send anything longer than PACKET_MAX_PAYLOAD_SIZE.  I'm wondering if PACKET_MAX_PAYLOAD_SIZE can be set to be equal to MAX_INPUT_LENGTH?

Reply:
(granted I see there's a size check on line 373, but it's still larger than inbuf

Reply:
So: I'm trying not to fix things in the mud at this point. the max payload size stuff is a protocol thing between doorman and xania: I don't think we should rely on the mud not overflowing based on it. We should fix that as part of #46 instead of hackily getting buffer sizes to have some kind of relationship in the code I think.

I made the minimum possible change here to at least have some agreed maximum size between xania and doorman without thinking too much: any suggestions what I should do here as part of this change?

Reply:
Made some changes per discussion in immchat; please take a look and lmk if you think this is OK as is for now, relying mostly on #46 to handle the input-from-doorman overflowing xania buffers side.